config/template.go

```diff
@@ -48,7 +82,12 @@ func defaultValue(defaultValue, templateValue interface{}) string {
 func NewTemplate(config []byte) (*Template, error) {
 	env := parseEnvironment(os.Environ())
 	tmpl, err := template.New("").Funcs(template.FuncMap{
-		"default": defaultValue,
+		"default":       defaultValue,
+		"split":         split,
+		"join":          join,
+		"replace":       replace,
+		"regexReplace":  regexReplace,
+		"regexpReplace": regexReplace,
```

Comment:
FYI, consul template uses `replaceAll` instead of `replace`, so if we are going to match, that one too?